modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationUtils.java

```diff
@@ -0,0 +1,506 @@
+                            if (policyContent.getBoolean(PluginConstants.PolicyConfigProperties.
+                                    POLICY_ENABLE) == (deviceFeature.isEnable())) {
+                                isCompliance = true;
+                                deviceFeature.setCompliance(isCompliance);
+                            } else {
+                                deviceFeature.setCompliance(isCompliance);
+                            }
+                            ComplianceFeature complianceFeature = setComplianceFeatures(activeFeature, deviceFeature);
+                            complianceFeatures.add(complianceFeature);
+                        }
+                        if (deviceFeature.getFeatureCode().equals(activeFeature.getFeatureCode()) &&
+                                deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
+                                        ENCRYPT_STORAGE)) {
+                            if (policyContent.getBoolean(PluginConstants.PolicyConfigProperties.
+                                    ENCRYPTED_ENABLE) == (deviceFeature.isEnable())) {
+                                isCompliance = true;
+                                deviceFeature.setCompliance(isCompliance);
+                            } else {
+                                deviceFeature.setCompliance(isCompliance);
+                            }
+                            ComplianceFeature complianceFeature = setComplianceFeatures(activeFeature, deviceFeature);
+                            complianceFeatures.add(complianceFeature);
+                        }
+                        if (deviceFeature.getFeatureCode().equals(activeFeature.getFeatureCode()) &&
+                                deviceFeature.getFeatureCode().equals(PluginConstants.OperationCodes.
+                                        PASSCODE_POLICY)) {
+                            if (policyContent.getBoolean(PluginConstants.PolicyConfigProperties.
+                                    ENABLE_PASSWORD) == (deviceFeature.isEnable())) {
+                                isCompliance = true;
+                                deviceFeature.setCompliance(isCompliance);
+                            } else {
+                                deviceFeature.setCompliance(isCompliance);
+                            }
+                            ComplianceFeature complianceFeature = setComplianceFeatures(activeFeature, deviceFeature);
+                            complianceFeatures.add(complianceFeature);
+                        }
+                    }
+                }
+                WindowsAPIUtils.getPolicyManagerService().checkPolicyCompliance(deviceIdentifier, complianceFeatures);
+            } catch (org.wso2.carbon.policy.mgt.common.FeatureManagementException e) {
+                String msg = "Error occurred while getting effective policy.";
```

Comment:
Avoid using "log and throw" as much as possible here as well as in all other relevant places.